integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -200,37 +242,54 @@ private void requestMasterContainer() throws Exception {
     if (!mMasterAddress.equals("localhost")) {
       relaxLocality = false;
     }
-    ContainerRequest masterContainerAsk =
-        new ContainerRequest(masterResource, nodes, null /* any racks */, PRIORITY, relaxLocality);
+    ContainerRequest masterContainerAsk = new ContainerRequest(masterResource, nodes,
+        null /* any racks */, MASTER_PRIORITY, relaxLocality);
     LOG.info("Making resource request for Tachyon master: cpu {} memory {} MB on node {}",
         masterResource.getVirtualCores(), masterResource.getMemory(), mMasterAddress);
     mRMClient.addContainerRequest(masterContainerAsk);
 
-    // Wait until Tachyon master container has been allocated
+    LOG.info("Waiting for master container to be allocated");
```

Comment:
since we are blocking here until the master container is allocated, I suggest to 
(1) update the javadoc of this method and indicate that the method will block until master container allocated; and
(2) who will notify the completion of allocation and when.